products/bash/validate-operator-releases.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+#******************************************************************************
+# Licensed Materials - Property of IBM
+# (c) Copyright IBM Corporation 2020. All Rights Reserved.
+#
+# Note to U.S. Government Users Restricted Rights:
+# Use, duplication or disclosure restricted by GSA ADP Schedule
+# Contract with IBM Corp.
+#******************************************************************************
+#
+# INSTRUCTIONS
+# ------------
+# To validate specific products, add the flag for your product, followed by the release name to the command line, e.g:
+#       ./validate-operator-releases.sh -m mq-demo -d ace-dashboard-demo -m mq-dev-demo
+#
+# Full parameter list:
+#   -n : CP4I Platform Navigator
+#   -m : MQ Queue Manager
+#   -d : ACE Dashboard
+#   -e : ACE Designer
+#   -r : CP4I Asset Repository
+
+function usage {
+    echo "Usage: $0 [products...]"
+}
+
+cp_releases=()
+
+while getopts "n:m:d:e:r:" opt; do
+  case ${opt} in
+    n ) cp_releases+=("$OPTARG^PlatformNavigator")
+      ;;
+    m ) cp_releases+=("$OPTARG^QueueManager")
+      ;;
+    d ) cp_releases+=("$OPTARG^Dashboard")
+      ;;
+    e ) cp_releases+=("$OPTARG^Designer")
+      ;;
+    r ) cp_releases+=("$OPTARG^AssetRepository")
+      ;;
+    \? ) usage
+      ;;
+  esac
+done
+
+if [[ -z "${cp_releases}" ]]; then
+    echo "No releases specified, validation complete."
+    exit 1
+fi
+
+function is_release_ready {
+  release_name=${1}
+  release_type=${2}
+  echo -e "\nChecking $release_name with type $release_type..."
+
+  release_status=$(oc get ${release_type} ${release_name} -o json | jq -r '.status.conditions')
```

Comment:
A couple of cases I don't think are handled. If there is no status or conditions element then this will return:
```
null
```
Which means not ready.

Also, the conditions array can contain multiple elements and it's not necessarily the 1st element that is the "Ready" one. I.e.:
```
[
  {
    "lastTransitionTime": "2020-07-08T15:01:03Z",
    "status": "False",
    "type": "Error"
  },
  {
    "lastTransitionTime": "2020-07-09T08:03:21Z",
    "message": "Platform Navigator has been provisioned.",
    "reason": "Deployed",
    "status": "True",
    "type": "Ready"
  }
]
```

Reply:
The null condition is checked by lines 58-60 with their empty check. The 3 scenarios I saw and explicitly checked are a return of null, a return of [] and a return of a set of data that I then check for the ready/true. 

Do you have a specific example of where the the conditions array gets multiple elements? Just so I can test as I go!